image_segmentation_node:
  ros__parameters:
    update_rate: 60.0
    debug_view: true

    rgb_topic: "/k4a/rgb/image_raw"
    depth_topic: "/k4a/depth_to_rgb/image_raw"
    use_compressed: false

    hsv_hue_low: 0
    hsv_hue_high: 179
    hsv_saturation_low: 0
    hsv_saturation_high: 255
    hsv_value_low: 0
    hsv_value_high: 100

    depth_low: 200
    depth_high: 430
    depth_scale: 1.0

    gaussian_kernel_size: [9, 9]
    gaussian_sigma: 5
    grayscale_threshold: 5
    morph_kernel_size: [3, 3]
    hsv_erode_iterations: 5
    depth_dilate_iterations: 100

    blob_min_threshold: 200
    blob_max_threshold: 1000
    blob_min_area: 50000
    blob_max_area: 10000000

point_cloud_generator_node:
  ros__parameters:
    update_rate: 60.0

    depth_scale: 1000.0                 
    voxel_size: 0.002                    

    tf_x: -0.236
    tf_y: -0.359
    tf_z: 0.1235
    tf_qx: -0.7068252
    tf_qy: 0.0
    tf_qz: 0.0
    tf_qw: 0.7073883